tests/unit/store/api.spec.js

```diff
@@ -0,0 +1,635 @@
+      const state = initialAPIState()
+      expect(state.backend).toEqual('SERVER')
+      expect(state.connected).toBe(true)
+      expect(state.connectionMessage).toBe(null)
+      expect(state.connectionTimeout).toBe(null)
+      expect(state.cloudUrl).toBe(undefined)
+      expect(state.retries).toBe(0)
+      expect(state.apiMode).toBe(null)
+      expect(state.serverUrl).toBe(undefined)
+      expect(state.version).toBe(null)
+    })
+    test('state checks local storage before environment variables', () => {
+      const state = localStoreAPIState()
+      expect(localStorage.getItem).toBeCalledWith('backend')
+      expect(state.backend).toEqual('foo')
+      expect(state.serverUrl).toBe('http://0.0.0.0:4200/graphql')
+    })
+    test('state checks environment variables if no local storage', () => {
+      const state = envVarAPIState()
+      expect(localStorage.getItem('backend')).toBe(null)
+      expect(state.backend).toEqual('CLOUD')
+      expect(state.cloudUrl).toBe('https://api.prefect.io/graphql')
+      expect(state.serverUrl).toBe('http://localhost:4200/graphql')
+    })
+  })
+
+  describe('getters when no local storage or environment variable is set', () => {
+    let store
+    beforeEach(() => {
+      const state = initialAPIState()
+      store = new Vuex.Store({
+        state: state,
+        getters: api.getters,
+        mutations: api.mutations
+      })
+    })
+    it('should return the backend', () => {
+      expect(store.getters.backend).toBe('SERVER')
+    })
+    it('should return connected state', () => {
+      expect(store.getters.connected).toBe(true)
+    })
+    it('should return connecting state', () => {
+      expect(store.getters.connecting).toBe(false)
+    })
+    it('should return the connection message', () => {
+      expect(store.getters.connectionMessage).toBe(null)
+    })
+    it('should return the connectionTimeout', () => {
+      expect(store.getters.connectionTimeout).toBe(null)
+    })
+    it('should return the api mode', () => {
+      expect(store.getters.apiMode).toBe(null)
+    })
+    it('should return a boolean about whether the backend is Cloud', () => {
+      expect(store.getters.isCloud).toBe(false)
+    })
+    it('should return a boolean about whether the backend is Server', () => {
+      expect(store.getters.isServer).toBe(true)
+    })
+    it('should return the cloud url', () => {
+      expect(store.getters.cloudUrl).toBe(undefined)
+    })
+    it('should return retries', () => {
+      expect(store.getters.retries).toBe(0)
+    })
+    it('should return the server url', () => {
+      expect(store.getters.serverUrl).toBe(undefined)
+    })
+    it('should return the active url', () => {
+      expect(store.getters.url).toBe(undefined)
+    })
+    it('should return the version', () => {
+      expect(store.getters.version).toBe(null)
+    })
+  })
+
+  describe('where relevant getters check local storage before environment variables', () => {
```

Comment:
I don't think this is a true statement - none of the getters for the API store check `localStorage` or environment variables; only the initial state is set based on those. As such, I think just the above `State`-testing describe block will cover this, meaning we can combine all 3 getters describe blocks into 1 general use pattern.